runtime/gc_glue_java/MarkingSchemeRootClearer.cpp

```diff
@@ -329,6 +330,27 @@ MM_MarkingSchemeRootClearer::scanContinuationObjects(MM_EnvironmentBase *env)
 						while (NULL != object) {
 							gcEnv->_markJavaStats._continuationCandidates += 1;
 							omrobjectptr_t next = _extensions->accessBarrier->getContinuationLink(object);
+#if defined(OMR_GC_CONCURRENT_SCAVENGER)
+							{
+								/**
+								 * In case of CS backout make sure the list contains the new version of the objects or if there is only one version,
+								 * due to being self-forwarded, restore the self-forwarded bit.
+								 * we could even check that CS backout was on before proceeding with this step, but probably not much savings anyway.
```

Comment:
actually, let's remove this line of the comment and put the runtime checks instead (I believe at this point backout flag has not been cleared yet)

_extensions->isConcurrentScavengerEnabled() && _extensions->isScavengerBackOutFlagRaised()